src/platforms/android/common/gradle.mdx

```diff
@@ -0,0 +1,240 @@
+    }
+}
+```
+
+```kotlin
+import io.sentry.android.gradle.InstrumentationFeature
+
+sentry {
+    // Disables or enables the handling of Proguard mapping for Sentry.
+    // If enabled the plugin will generate a UUID and will take care of
+    // uploading the mapping to Sentry. If disabled, all the logic
+    // related to proguard mapping will be excluded.
+    // Default is enabled.
+    includeProguardMapping.set(true)
+    
+    // Whether the plugin should attempt to auto-upload the mapping file to Sentry or not.
+    // If disabled the plugin will run a dry-run and just generate a UUID.
+    // The mapping file has to be uploaded manually via sentry-cli in this case.
+    // Default is enabled.
+    autoUploadProguardMapping.set(true)
+
+    // Disables or enables the automatic configuration of Native Symbols
+    // for Sentry. This executes sentry-cli automatically so
+    // you don't need to do it manually.
+    // Default is disabled.
+    uploadNativeSymbols.set(false)
+
+    // Does or doesn't include the source code of native code for Sentry.
+    // This executes sentry-cli with the --include-sources param. automatically so
+    // you don't need to do it manually.
+    // Default is disabled.
+    includeNativeSources.set(false)
+
+    // Enable or disable the tracing instrumentation.
+    // Does auto instrumentation for specified features through bytecode manipulation.
+    // Default is enabled.
+    // Only available v3.0.0 and above.
+    tracingInstrumentation {
+      enabled.set(true)
+      
+      // Specifies a set of instrumentation features that are eligible for bytecode manipulation.
+      // Defaults to all available values of InstrumentationFeature enum class.
+      features.set(setOf(InstrumentationFeature.DATABASE, InstrumentationFeature.FILE_IO))
+    }
+}
+```
+
+Now when you build your app, the plugin will upload the ProGuard/R8 mappings, source bundle, and native symbols as you configured them to Sentry.
+
+### Variant Filtering
+
+You can specify which variant/flavor/build-type should be ignored by Sentry in your `app/build.gradle` file:
+
+```groovy
+sentry {
+    // List the build types that should be ignored (e.g. "release").
+    ignoredBuildTypes = ["release"]
+
+    // List the build flavors that should be ignored (e.g. "production").
+    ignoredFlavors = ["production"]
+
+    // List the build variant that should be ignored (e.g. "productionRelease").
+    ignoredVariants = ["productionRelease"]
+}
+```
+
+```kotlin
+sentry {
+    // List the build types that should be ignored (e.g. "release").
+    ignoredBuildTypes.set(setOf("release"))
+
+    // List the build flavors that should be ignored (e.g. "production").
+    ignoredFlavors.set(setOf("production"))
+
+    // List the build variant that should be ignored (e.g. "productionRelease").
+    ignoredVariants.set(setOf("productionRelease"))
+}
+```
+
+## ProGuard/R8
```

Comment:
The Proguard specific info could go into an `include` and we could use it here, and also on the ProGuard page to make sure content doesn't go out of sync

Reply:
mm, but on the proguard page we just redirect to either Gradle integration or sentry-cli, there's no specifics about how it works, so maybe it's fine like that?